tests/unit_test.py

```diff
@@ -0,0 +1,572 @@
+    ('nm^{-1}', _Pow(nanometer, -1)),
+    ('rad/s', _Mul(rad, _Pow(second, -1))),
+    ('rad/sec', _Mul(rad, _Pow(second, -1))),
+    ('rad\\: /\\: sec', _Mul(rad, _Pow(second, -1))),
+    ('s^{-1}', _Pow(second, -1)),
+    ('slugs/ft3', _Mul(slug, _Pow(_Mul(foot, 3), -1))),
+    ('V/us', _Mul(volt, _Pow(microsecond, -1))),
+    ('V\\cdot m', _Mul(volt, meter)),
+    ('kJ/mol', _Mul(create_prefixed_unit(joule, SI_PREFIXES['k']), _Pow(mole, -1))),
+    ('kN/m', _Mul(create_prefixed_unit(newton, SI_PREFIXES['k']), _Pow(meter, -1))),
+    ('molar*ohm', _Mul(molar, ohm)),
+    ('M*\\Omega', _Mul(molar, ohm)),
+]
+
+
+@pytest.mark.parametrize('input, output', unit_examples)
+def test_parse_as_unit_should_succeed(input, output):
+    assert_equal(input, output, parse_as_unit=True)
+
+
+bad_unit_examples = [
+    # non-Quantity, numeric value or symbols
+    '1',
+    'E-9',
+    'E3',
+    'x+y',
+    '\\frac{L_{1}}{R_{total}}',
+    'apples',
+    'apples\\times grams',
+    'pen\\times pineapple\\times apple\\times pen',
+    'red\\: delicious',
+    'carbon\\: atoms',
+    'Pokeballs\\: with\\: her\\: income',
+    'year\\left(s\\right)\\: of\\: her\\: life\\: due\\: to\\: bickering',
+
+    # prefix + non-Quantity
+    '\\mu 1',
+    '\\mu x',
+
+    # invalid prefix + quantity
+    '\\mu slug',
+
+    # unsupported constant or prefix w/o Quantity
+    'G',
+    'giga',
+    'pico',
+
+    # combined prefix with non-abbreviated name
+    'Mohms',  # megaohms or M\\Omega
+    'kohms',  # killiohms or k\\Omega
+    'kg/kmole',  # kmol
+    'mb',  # millibar or mbar
+
+    # operatorname and other latex cmds
+    '\\$',
+    '\\operatorname{cm}',
+    '\\operatorname{kg}',
+    '\\frac{J}{\\operatorname{kg}K}',
+    '\\sec ',
+    '\\min ',
+
+    # letters or subscripts after valid units
+    'A\\: peak',
+    'ADC',
+    'Ap',
+    'Hz\\: \\left(10\\%\\: tolerance\\: for\\: this\\: answer\\right)',
+    'V_{pp}',
+    'V_{RMS}',
+    'Vpeak',
+    'Vrms',
+
+    # unit phrases
+    'cubic\\: inches',
+    'meters\\: per\\: second',
+    'meters\\: per\\: second\\: squared',
+    'miles\\: per\\: hour',
+    'square\\: inches',
+
+    # unicode
+    'kΩ',
+    'º',
+    '°C\\: ',
+    'Ω',
+
+    # valid unit combined with invalid unit
+    'MeV/nucleon',
+
+    # assorted invalid units
+    '\\degree s',  # plural degrees w/ latex
+    'o',  # meant to be \degree
+    'msec',  # ms
+
+    # unsupported CGS units
+    'dyne',
+    'erg',
+    'statampere',
+    'statcoulomb',
+    'statvolt',
+    'gauss',
+    'maxwell',
+    'debye',
+    'oersted',
+
+    # other unsupported units
+    'quart',
+    'planck',
+
+    'ounce',
+    'oz',
+    'gallon',
+    'decade',
+    'octave',
+    'R',
+    'dbV',
+    'gpm',
+    'hp',
+
+    'lbf⋅ft',
+    'lbf.ft',
+    'lb⋅in',
+    'lb.in',
+    'ft⋅lbf',
+    'ft.lbf',
+    'ft⋅lb',
+    'ft.lb',
+
+    'psia',
+
+    'PPS',  # pulses per second
+    'kN.m',
```

Comment:
1. What unit is this?
2. What are we going to do for correctness with periods? Outside of scope for the parser, but if the student doesn't have a period but the instructor did, do we consider it correct or not?

Reply:
Or wait, this is the unsupported list, was the decision we aren't supporting periods? Totally cool with that, just trying to remember

Reply:
@lily514 Yeah, we decided to keep it as is. So we will throw an exception

Reply:
@lily514 this was from a Suffix in the DB

we are not allowing periods in either the instructor unit or the student unit